packages/composites/market-status/src/transport/market-status.ts

```diff
@@ -0,0 +1,163 @@
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { makeLogger, sleep } from '@chainlink/external-adapter-framework/util'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { AdapterResponse } from '@chainlink/external-adapter-framework/util/types'
+
+import { AdapterName, marketAdapters } from '../config/adapters'
+import { inputParameters } from '../endpoint/market-status'
+import type { BaseEndpointTypes } from '../endpoint/market-status'
+
+const logger = makeLogger('MarketStatusTransport')
+
+type MarketStatusResult = {
+  marketStatus: MarketStatus
+  providerIndicatedTimeUnixMs: number
+  source?: AdapterName
+}
+
+type RequestParams = typeof inputParameters.validated
+
+export class MarketStatusTransport extends SubscriptionTransport<BaseEndpointTypes> {
+  requester!: Requester
+
+  async initialize(
+    dependencies: TransportDependencies<BaseEndpointTypes>,
+    adapterSettings: BaseEndpointTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.requester = dependencies.requester
+  }
+
+  async backgroundHandler(context: EndpointContext<BaseEndpointTypes>, entries: RequestParams[]) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(context, param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(context: EndpointContext<BaseEndpointTypes>, param: RequestParams) {
+    const requestedAt = Date.now()
+
+    let response: AdapterResponse<BaseEndpointTypes['Response']>
+    try {
+      const result = await this._handleRequest(context, param)
+      response = {
+        data: {
+          result: result.marketStatus,
+          source: result.source,
+        },
+        result: result.marketStatus,
+        statusCode: 200,
+        timestamps: {
+          providerDataRequestedUnixMs: requestedAt,
+          providerDataReceivedUnixMs: Date.now(),
+          providerIndicatedTimeUnixMs: result.providerIndicatedTimeUnixMs,
+        },
+      }
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : `Unknown error occurred: ${e}`
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: requestedAt,
+          providerDataReceivedUnixMs: Date.now(),
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequest(
+    context: EndpointContext<BaseEndpointTypes>,
+    param: RequestParams,
+  ): Promise<MarketStatusResult> {
+    const market = param.market
+    if (!market) {
+      throw new Error(`Missing market in params: ${market}`)
+    }
```

Comment:
We should validate this on layer up. either in InputParameters or use customInputValidation

Reply:
`market` is [required](https://github.com/smartcontractkit/ea-framework-js/blob/main/src/adapter/market-status.ts#L25) input param. The validation for required input params are handled internally in the ea-framework, so this if statement can be removed